### DISCUSSION TITLE
Something wrong while generating license.svg

### DISCUSSION THREAD

**Discussion by GeorgeDong32** [lowlighter/metrics#1226] [Q&A] [answered]
While I'm using metrics to generate a license svg, there seem to be something wrong. 
First was that the npm ci setup occurred an error

![image](https://user-images.githubusercontent.com/98630204/189520610-d47ea98e-ecb8-4619-acb4-fd12e766bee5.png)

The second was the saving process error

![image](https://user-images.githubusercontent.com/98630204/189520765-140251fd-cd37-422e-ada9-570e0b83efe0.png)

**Comment by GeorgeDong32** [comment]
With my workflows yml

![image](https://user-images.githubusercontent.com/98630204/189520851-8a4c8732-ebee-4dea-8b99-56704e9666e6.png)

**Comment by GeorgeDong32** [comment, marked as answer]
I have solved it by removing the command "plugin_licenses_setup: npm ci" and adding permission "write". Hope it can help some of you.

**Reply by lowlighter** [reply]
Glad you found a way to make it work 😅 !

I think maybe the default command should be changed to something like `[[ -f package.json ]] && npm ci` to stop assuming that every project setup is a NodeJS one and avoid confusing errors like this one when it tries to do something that come from another ecosystem

Edit: Actually it seems to be already the case, maybe it means the doc isn't clear enough 🤔 Should probably patch it to make it clearer that this option is intended to be run for projects that require setup/build steps and is optional if not needed

**Reply by GeorgeDong32** [reply]
Thank you for replying to me. The documentations may need to be updated, and if you need help, I'm happy to help. 😃
